operator/pkg/remote/kyma.go

```diff
@@ -142,25 +142,33 @@ func (c *KymaSynchronizationContext) SynchronizeRemoteKyma(ctx context.Context,
 		controllerutil.AddFinalizer(remoteKyma, labels.Finalizer)
 	}
 
+	previousRemoteObservedGeneration := remoteKyma.Status.ObservedGeneration
 	if remoteKyma.Status.ObservedGeneration != remoteKyma.GetGeneration() {
+		remoteKyma.Status.ObservedGeneration = remoteKyma.GetGeneration()
+		if err := c.runtimeClient.Status().Update(ctx, remoteKyma); err != nil {
+			return false, err
+		}
+
 		kyma.Spec = remoteKyma.Spec
 		err := c.controlPlaneClient.Update(ctx, kyma)
 		if err != nil {
-			recorder.Event(remoteKyma, "Warning", err.Error(), "Client could not update Control Plane Kyma")
-			return true, err
-		}
-		remoteKyma.Status.ObservedGeneration = remoteKyma.GetGeneration()
-		err = c.runtimeClient.Status().Update(ctx, remoteKyma)
-		if err != nil {
-			return true, err
+			// revert remote observed generation
+			remoteKyma.Status.ObservedGeneration = previousRemoteObservedGeneration
+			if err := c.runtimeClient.Status().Update(ctx, remoteKyma); err != nil {
+				return false, err
+			}
+			recorder.Event(remoteKyma, "Warning", err.Error(), "Client could not update "+
+				"Control Plane Kyma while syncing with remote Kyma")
```

Comment:
I think this logic is covered with a change in https://github.com/kyma-project/kyma-operator/pull/58/files#diff-bc1bb91f1f01617c2d3693314155dcfad6e5a612788e40ec921f8236d0a5a11cR144-R155

Reply:
will remove it as discussed, so that you address it in your PR :)